pkg/util/metamorphic/metamorphicutil/is_metamorphic.go

```diff
@@ -0,0 +1,23 @@
+// Copyright 2024 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package metamorphicutil
```

Comment:
Nit: `util.metamorphic.metamorphicutil` is kinda redundant... if we don't immediately expect other additions to this package, maybe `buildflags` is a better name?

Reply:
I'm not 100% clear on the details, but I do expect other additions to this package. I think that we will probably add some sort of `map[string]interface{}` as a "catalog" of the metamorphic variables. So `buildflags` I don't think will be appropriate. Won't be hard to change the name later if a better one is found while we're adding the new features.